### DISCUSSION TITLE
Can I know how to print the supergraph Schema with Routing Mechanisms

### DISCUSSION THREAD

**Discussion by shamin2021** [graph-quilt/graphql-orchestrator-java#194] [Q&A]
GraphQLOrchestrator graphQLOrchestrator = GraphQLOrchestrator.newOrchestrator()
                .runtimeGraph(runtimeGraph).build();
        String orchestrate = new SchemaPrinter().print(runtimeGraph.getExecutableSchema());
        System.out.println(orchestrate);

I have tried this and has given me a schemas as the following 
"[bookService]"
type Author {
  firstName: String!
  id: String
  lastName: String!
}

"[bookService]"
type Book {
  author: Author
  id: ID
  name: String
  pageCount: Int
}

"[]"
type Query {
  _namespace: String
  authorByid(id: ID!): Author
  bookById(id: ID): Book
}

"A selection set"
scalar _FieldSet
would love to know if theres a way to print the routing mechanisms as well

**Comment by ashpak-shaikh** [comment]
We don't print the routing mechanisms at this point, but you can print the outgoing queries in the query method of the ServiceProvider interface.

**Reply by shamin2021** [reply]
Hi, can you please help me achieve this following scenario 
I have two endpoints http://localhost:9092/ & http://localhost:9091/ with two graphql services where the schemas are annotated with federation directives , i would like to stitch these using orcestrator and have started implementation as a test case in graphql gateway java 
import com.intuit.graphql.gateway.registry.SdlServiceRegistration;
import com.intuit.graphql.orchestrator.ServiceProvider;
import com.intuit.graphql.gateway.provider.SdlServiceProvider;
import com.intuit.graphql.gateway.registry.ServiceDefinition;
import com.intuit.graphql.gateway.registry.ServiceRegistration;
import com.intuit.graphql.orchestrator.batch.BatchLoaderExecutionHooks;
import com.intuit.graphql.orchestrator.schema.RuntimeGraph;
import com.intuit.graphql.orchestrator.stitching.SchemaStitcher;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.concurrent.ExecutionException;

public class Test3 {

    public static void main(String[] args) throws ExecutionException, InterruptedException {

        // Define Service Definitions
        ServiceDefinition sd1 = ServiceDefinition.newBuilder()
                .namespace("service1")
                .endpoint("http://localhost:9091/graphql")
                .type(ServiceDefinition.Type.GRAPHQL)
                .build();

        ServiceDefinition sd2 = ServiceDefinition.newBuilder()
                .namespace("service2")
                .endpoint("http://localhost:9092/graphql")  // Adjusted endpoint for the second service
                .type(ServiceDefinition.Type.GRAPHQL)
                .build();


        // Create SDL service registrations for each endpoint
        SdlServiceRegistration serviceRegistration1 = new SdlServiceRegistration(sd1);
        SdlServiceRegistration serviceRegistration2 = new SdlServiceRegistration(sd2);

        // Create service providers for each endpoint using SDL service registrations
        SdlServiceProvider serviceProvider1 = new SdlServiceProvider(serviceRegistration1);
        SdlServiceProvider serviceProvider2 = new SdlServiceProvider(serviceRegistration2);

        // Stitch together schemas from service providers
        RuntimeGraph runtimeGraph = SchemaStitcher.newBuilder()
                .service(serviceProvider1)
                .service(serviceProvider2)
                .build()
                .stitchGraph();

        // Obtain the unified schema from the stitched graph
        String unifiedSchema = runtimeGraph.getExecutableSchema().toString();

        // Print the unified schema
        System.out.println(unifiedSchema);
    }
}
Can you please help me with the implementation
